src/NBomber/DomainServices/Reports/ConsoleReport.fs

```diff
@@ -70,6 +70,16 @@ module ConsoleNodeStats =
         [ ConsoleStatusCodesStats.printScenarioHeader scnStats.ScenarioName
           ConsoleStatusCodesStats.printStatusCodeTable scnStats ]
 
+    let private printThresholdStats (scnStats: ScenarioStats) =
+        let headers = ["threshold"; "status"]
+        let rows =
+            scnStats.ThresholdStats
+            |> Option.map (ReportHelper.ThresholdStats.createTableRows Console.okEscColor Console.errorEscColor)
+            |> Option.defaultValue List.empty
+
+        [ Console.addLine $"thresholds for scenario: {Console.okColor scnStats.ScenarioName}"
+          Console.addTable headers rows ]
```

Comment:
That's how it looks like:
![image](https://user-images.githubusercontent.com/17567634/169595606-d2cc7095-7b4e-43c3-8479-c9c4ecbd493e.png)

I'm not sure if this would be sufficient, and if not, I'm not sure at the moment how we could print a string representation for the thresholds predicates.

Reply:
For F# we could use https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/code-quotations

Reply:
Also, we could have an additional API that will accept the optional string argument that the user will fill, and it will be printed if it's not empty. But maybe, for now, we can leave all this "Fetching Meta Info" machinery as for v2?

Reply:
Hi @pavlogrushetsky 
You can take a look how this library is working for F#
https://github.com/SwensenSoftware/unquote

Reply:
Thank you! I will take a look.

Reply:
@AntyaDev It looks like I could utilize `Unquote` library. I've played with it a little bit, and was able to specify a threshold using quotation expression. The work is still in progress, but I think, I would need to separate `Threshold` model to have on in `Contracts` with function union cases (like `int -> bool`), and one in `DomainTypes` with `Quotations.Expr` union cases (like `Quotations.Expr<int -> bool>`). This would allow us to keep the consumer code as simple as possible, to not to have something like the following:

```fsharp
|> Scenario.withThresholds [
    RequestAllCount(<@ fun x -> x = 3 @>)
]
```

I would just need to translate `Contracts.Threshold` model to `DomainTypes.Threshold` model, like you do for `Scenario`.

@AntyaDev Please, let me know if this makes sense to you.

Reply:
Yes, it sounds reasonable!

Reply:
Looks very good, I like how F# and C# look like.
Btw what was the issue with F# quotations? I thought that you would not be able to use it for C# and to use it for F#, you should wrap the predicate into <@ predicate @>

Reply:
> Btw what was the issue with F# quotations? I thought that you would not be able to use it for C# and to use it for F#, you should wrap the predicate into <@ predicate @>

Yep, I was able to wrap the predicates in `<@ @>` in the domain logic and keep using just plain lambdas like `fun x -> x > 10` in both C# and F#, and the thresholds evaluation logic worked good. And main reason to do this was to get a string representation for those quotation expressions, but it turned out that it is not really human readable when I tried to de-compile them.